charts/newrelic-infrastructure-beta/templates/control-plane/daemonset.yaml

```diff
@@ -0,0 +1,213 @@
+            {{- end }}
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "newrelic.licenseSecretName" . | required "licenseKey or customSecretName and customSecretLicenseKey must be set" }}
+                  key: {{ include "newrelic.licenseSecretKey" . | required "licenseKey or customSecretName and customSecretLicenseKey must be set" }}
+            {{- if (include "newrelic.nrStaging" .) }}
+            - name: NRIA_STAGING
+              value: "true"
+            {{- end }}
+            {{- with .Values.eventQueueDepth }}
+            - name: "NRIA_EVENT_QUEUE_DEPTH"
+              value: {{ .| quote }}
+            {{- end }}
+            # Using FORWARD_ONLY mode to avoid the creating a host entity for the pod.
+            - name: NRIA_IS_SECURE_FORWARD_ONLY
+              value: "false"
+            - name: NRIA_IS_FORWARD_ONLY
+              value: "true"
+            {{- with .Values.controlPlane.extraEnv }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          {{- with .Values.controlPlane.extraEnvFrom }}
+          envFrom:
+            {{- tpl . $ | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - mountPath: /var/db/newrelic-infra/data
+              name: tmpfs-data
+            - mountPath: /var/db/newrelic-infra/user_data
+              name: tmpfs-user-data
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            {{- if .Values.config }}
+            - name: config
+              mountPath: /etc/newrelic-infra.yml
+              subPath: newrelic-infra.yml
+            {{- end }}
+            {{- with .Values.controlPlane.extraVolumeMounts }}
+            {{- tpl . $ | nindent 12 }}
+            {{- end }}
+          {{- with .Values.controlPlane.resources }}
+          resources: {{ toYaml . | nindent 12 }}
+          {{- end }}
+      volumes:
+        - name: nri-kubernetes-config
+          configMap:
+            name: {{ template "newrelic.fullname" . }}-control-plane
+            items:
+              - key: nri-kubernetes-config.yml
+                path: nri-kubernetes-config.yml
+        - name: tmpfs-data
+          emptyDir: {}
+        - name: tmpfs-user-data
+          emptyDir: {}
+        - name: tmpfs-tmp
+          emptyDir: {}
+        {{- if .Values.config }}
+        - name: config
+          configMap:
+            name: {{ template "newrelic.fullname" . }}
+            items:
+              - key: newrelic-infra.yml
+                path: newrelic-infra.yml
+        {{- end }}
+        {{- with .Values.controlPlane.extraVolumes }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . }}
+      {{- end }}
+      {{- with .Values.controlPlane.affinity.nodeAffinity }}
+      affinity:
+        nodeAffinity: {{ . | toYaml | nindent 10 }}
```

Comment:
```suggestion
        nodeAffinity:
          {{- . | toYaml | nindent 10 }}
```

Reply:
Are we sure the `affinity` key is going to exist and not return a `nil` and break the templating?